File: Posts/Data-Strucutres-And-Algorithms/Prime-Number/Sieve-Of-Eratosthenes.cpp
#include <iostream>
#include <math.h>
#include <vector>

using namespace std;

void sieveOfEratosthenes() {
    int n = (int)1e6;
    vector < bool > is_prime(n + 1, true);
    is_prime[0] = is_prime[1] = false;
    for (int p = 2; p <= n; ++p) {
        if (is_prime[p] == true && (long long)p * p <= n) {
            for (int i = p * p; i <= n; i += p) {
                is_prime[i] = false;
            }
        }
    }
    for (int i = 1; i <= n; ++i) {
        if (is_prime[i] == true) {
            cout << i << " ";
        }
    }
    cout << endl;
}

int main() {
    sieveOfEratosthenes();
    return 0;
}

File: osts/data-strucutres-and-algorithms/prime-number/Sieve-Of-Eratosthenes.py
import math


def sieveOfEratosthenes():
    n = 1000000
    is_prime = [True for i in range(0, n + 1)]
    
    is_prime[0] = is_prime[1] = False
    
    for p in range(2, n + 1):
        if is_prime[p] == True and p * p <= n:
            for i in range(p * p, n + 1, p):
                is_prime[i] = False
    
    for i in range(1, n + 1):
        if is_prime[i] == True:
            print(i)


sieveOfEratosthenes()
